src/camera.py

```diff
@@ -0,0 +1,181 @@
+"""
+camera
+
+Author: Steven Dropiewski
+"""
+import numpy as np
+import time
+
+from picar import filedb
+from picar.SunFounder_PCA9685 import Servo
+import picar
+
+
+class Camera:
+    """
+    Creates a camera object that can be panned and tilted.
+
+    The camera can pan in [0, 180] degrees and can tilt in [0, 150] degrees.
+
+    :param bus_number: The bus number of the servos.
+    :type bus_number: int
+
+    :param pan_channel: Pan servo's I2C channel.
+    :type pan_channel: int
+
+    :param tilt_channel: Tilt servo's I2C channel.
+    :type tilt_channel: int
+
+    :param offset: Servo's offset.
+    :type offset: int
+    """
+
+    MAX_PAN_DEGREE = 180
+    """The maximum value to be written to pan servo."""
+    MAX_TILT_DEGREE = 150
+    """The maximum value to be written to tilt servo."""
+    MIN_DEGREE = 0
+    """The minimum value to be written to both pan and tilt servo."""
+    MAX_PAN_CALI_DEGREE = 180
+    """The maximum value to calibrate pan."""
+    MAX_TILT_CALI_DEGREE = 150
+    """The maximum value to calibrate tilt."""
+    MIN_PAN_CALI_DEGREE = -180
+    """The minimum value to calibrate pan."""
+    MIN_TILT_CALI_DEGREE = -150
+    """The minimum value to calibrate tilt."""
+
+    def __init__(self, bus_number=1, pan_channel = 1, tilt_channel = 0, offset = 0):
+        picar.setup()
+
+        self._db = filedb.fileDB("config")
+
+        self._pan_offset = int(self._DB.get('pan_offset', default_value=0))
+        self._tilt_offset = int(self._DB.get('tilt_offset', default_value=0))
+
+        self._pan_servo = Servo.Servo(pan_channel, bus_number=bus_number,
+                                                   offset=offset)
+        self._tilt_servo = Servo.Servo(tilt_channel, bus_number=bus_number,
+                                                     offset=offset)
+
+        self._current_pan_angle = 0
+        self._current_tilt_angle = 0
+
+
+    def pan(self, degree):
+        """
+        Pans the camera.
+
+        Controls the horizontal movement of the camera in [0, 180] degrees.
+        0 refers to the vehicle's right while 180 refers to the vehicle's left.
+        The degrees will be clamped to [0, 180].
+
+        :param degree: The number of degrees.
+        :type degree: int
+        """
+        self._current_pan_angle = np.clip(degree, self.MIN_DEGREE,
+                                                  self.MAX_PAN_DEGREE)
+        self._pan_servo.write(self._current_pan_angle + self._pan_offset)
```

Comment:
Shouldn't this be in the clip validation above? See this example:
```python
self.pan(180)
# self._current_pan_angle is 180
# self._pan_offset > 0, say 10
# self._pan_servo.write(190)
```

Reply:
Also if
```python
self.pan(0)
```
with a negative pan offset.

Reply:
That's why I had it validate twice. Once when setting current pan angle and then before writing to the servo. Otherwise if you give it 90, but you have changed the calibration so 90 is actually 85. _current_pan_angle will then show 85, when it's actually at 90, or what you want to refer to as 90.